torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2122,6 +2187,55 @@ def __init__(
             # This is needed when 'deepcopy'-ing this object.
             self.graph.inserting_before(self.graph.output(None))
 
+            # Mapping of each node name to the node itself.
+            #
+            # This is useful for matching an FX node from a recorded ShapeEnv.graph
+            # to the FX node of the ShapeEnv we are running the event on.
+            #
+            # Whenever you add a node to self.graph, you must add a mapping to this
+            # variable. Otherwise, the built FX graph on the replayed ShapeEnv will
+            # not be valid.
+            self.name_to_node: Dict[str, torch.fx.Node] = {}
+
+    def check_equal(self, other: "ShapeEnv") -> None:
+        shape_env_check_state_equal(self, other)
```

Comment:
it's weird this is a top-level function lol

Reply:
I did it just so I could move this implementation somewhere else. Would you rather have it inlined here?

Reply:
Meh it's probably not worth changing. The competing concerns is having the definition close to the class so that you remember to update it vs the class already being really long

Reply:
Hmmm. Maybe I could make those parts that will probably have to be updated into parameters. Specifically:

- `non_state_variable_names`: list of fields that should not be compared
- `map_value`: function that maps the value of each state variable to its comparable part

What do you think?